src/python/py/models/builder.py

```diff
@@ -509,22 +523,36 @@ def order_repeated_field(self, repeated_proto, key_name, order):
         order = list(order)
         repeated_proto.sort(key=lambda x: order.index(getattr(x, key_name)))
 
-    def make_external_tensor(self, np_data, name, unpack_int4=False, **kwargs):
-        tensor = numpy_helper.from_array(np_data)
-        tensor.name = name
+    def make_tensor_proto_from_tensor(self, tensor, name):
+        raw_data = bytes(
```

Comment:
`bytes(...)` makes copy. And I am guessing helper.make_tensor makes another copy of the provided bytes? Hopefully not. Anyway, I don't think there is any good alternative as of now given this code is all written in Python.

But good work trying to avoid making copies so far.